docs/testplan/ecn/WRED-and-ECN-statistics-testplan.md

```diff
@@ -0,0 +1,254 @@
+        show interfaces counters detailed Ethernet&lt;&gt;
+
+
+
+
+ Test cases
+ ----------
+ WRED/ECN statistics verification on supported platform
+
+ Terminology:<br>
+    WRED=0/1 refers WRED disabled/enabled on queue <br>
+    ECN=0/1 refers ECN disabled/enabled queue <br>
+    CODE POINT refers to ECN CODE POINT in the IP header <br>
+ | <div style="width:250px">Testcase</div> | <div style="width:300px">Test expectation </div>                                               | <div style="width:500px">Test Steps</div> |
+ | --------------------------------------- | -----------------------------------------------------------------------------------------------| ----------------------------------------- |
+ | WRED=0 ECN=0 CODE POINT=XX              | Tail Drop and hence no WRED/ECN counter increments                                                                 | 1. Query STATE_DB for WRED/ECN support on the platform <br> 2. tx disable egress port  <br>    3. Send TCP/UDP traffic. <br> 4. tx enable egress port <br> 5. Verify WRED/ECN packet counter and Byte counter <br>        |
+ | WRED=1 ECN=0 CODE POINT=XX              | If WRED Algorithm determines pkt is drop eligible then increment WRED counters. ECN counters should not increment . | 1. Query STATE_DB for WRED/ECN support on the platform <br> 2. tx disable egress port  <br>    3. Send TCP/UDP traffic. <br> 4. tx enable egress port <br> 5. Verify WRED/ECN packet counter and Byte counter <br>  |
+ | WRED=1 ECN=1 CODE POINT=00              | If WRED Algorithm determines pkt is drop eligible then increment WRED counters. ECN counters should not increment . | 1. Query STATE_DB for WRED/ECN support on the platform <br> 2. tx disable egress port  <br>    3. Send TCP/UDP traffic. <br> 4. tx enable egress port <br> 5. Verify WRED/ECN packet counter and Byte counter <br> |
+ | WRED=1 ECN=1 CODE POINT=01              | If WRED Algorithm determines pkt is drop eligible then increment ECN counters. WRED counters should not increment . | 1. Query STATE_DB for WRED/ECN support on the platform <br> 2. tx disable  egress port  <br>    3. Send TCP/UDP traffic. <br> 4. tx enable egress port <br> 5. Verify WRED/ECN packet counter and Byte counter <br> |
+ | WRED=1 ECN=1 CODE POINT=10              | If WRED Algorithm determines pkt is drop eligible then increment ECN counters. WRED counters should not increment . | 1. Query STATE_DB for WRED/ECN support on the platform <br> 2. tx disable  egress port  <br>    3. Send TCP/UDP traffic. <br> 4. tx enable egress port <br> 5. Verify WRED/ECN packet counter and Byte counter <br> |
+ | WRED=1 ECN=1 CODE POINT=11              | If WRED Algorithm determines pkt is drop eligible then increment ECN counters. WRED counters should not increment . | 1. Query STATE_DB for WRED/ECN support on the platform <br> 2. tx disable egress port  <br>    3. Send TCP/UDP traffic. <br> 4. tx enable egress port <br> 5. Verify WRED/ECN packet counter and Byte counter <br> |
```

Comment:
When WRED=1 ECN=1 CODE POINT=11, Can you double-check the Test expectation?
In my opinion, WRED Algorithm will not drop the packet and mark the packet, so, ECN counters and WRED counters should not increment.